### PR TITLE
perf(ext/http): Migrate op_http_get_request_method_and_url to v8::Array

### PR DESCRIPTION
**Pull request by kamilogorek**
~This PR is based on work from https://github.com/denoland/deno/pull/19354 and only 2nd commit is relevant. I'll rebase this PR once the initial commit that introduced `smallvec` will be merged.~

Tackles 3rd item from https://github.com/denoland/deno/issues/19330 list.

cc @mmastrac

### CODE REVIEW

File: ext/http/http_next.rs [hunk 1]
```diff
@@ -231,20 +232,56 @@ where
...
   )
+  .unwrap()
+  .into();
+
+  let port: v8::Local<v8::Value> = match request_info.peer_port {
+    Some(port) => v8::Integer::new(scope, port.into()).into(),
+    None => v8::undefined(scope).into(),
+  };
+
+  let vec: SmallVec<[v8::Local<v8::Value>; 5]> =
```
Comment (kamilogorek) [inline ext/http/http_next.rs:278]: As we know the exact size beforehand, can we somehow create `v8::Array` directly, skipping `SmallVec` part here?

Comment (mmastrac) [reply ext/http/http_next.rs:278]: In this case, I recommend creating a Rust array of 5 elements instead of using `SmallVec`.

Comment (kamilogorek) [reply ext/http/http_next.rs:278]: Looking at the docs, this should also do the trick I think?

```rs
let array = v8::Array::new(scope, 5);
array.set_index(scope, 0, method);
array.set_index(scope, 1, authority);
array.set_index(scope, 2, path);
array.set_index(scope, 3, peer_address);
array.set_index(scope, 4, port);
```

Comment (kamilogorek) [reply ext/http/http_next.rs]: I ended up using a simple array, as it's fixed-size anyway so we don't need vec here either.

**Comment by kamilogorek** [comment]
Rebased and left the only relevant commit.

**Comment by mmastrac** [comment]
Great! I'll run a bench w/this as soon as I can, probably later this weekend. Thanks for the contributions.

**Comment by mmastrac** [comment]
The old test case wasn't exercising this path, but I created one that does.

Before: 113.9k
After: 114.3k

Small change but in the right direction! Thanks again.

**Review by mmastrac — APPROVED** [review]
LGTM

**Comment by kamilogorek** [comment]
1% here, 1% there and all of a sudden it's 5-10% improvement :)

**Comment by mmastrac** [comment]
Definitely! There are no quick wins just lots of little ones that add up 😄

**Pull request merged**